loky/backend/utils.py

```diff
@@ -84,3 +94,207 @@ def _recursive_terminate(pid):
             # level function raise a warning and retry to kill the process.
             if e.errno != errno.ESRCH:
                 raise
+
+
+##############################################################################
+# The following code is derived from code by Intel developper @anton-malakhov
+# available at https://github.com/IntelPython/smp
+#
+# Copyright (c) 2017, Intel Corporation published under the BSD 3-Clause
+# license
+#
+
+
+# Structure to cast the the info on dynamically loaded library.
+if sys.maxsize > 2**32:
+    class dl_phdr_info(ctypes.Structure):
+        _fields_ = [("dlpi_addr",  ctypes.c_uint64),
+                    ("dlpi_name",  ctypes.c_char_p),
+                    ("dlpi_phdr",  ctypes.c_void_p),
+                    ("dlpi_phnum", ctypes.c_uint16)]
+else:
+    class dl_phdr_info(ctypes.Structure):
+        _fields_ = [("dlpi_addr",  ctypes.c_uint32),
+                    ("dlpi_name",  ctypes.c_char_p),
+                    ("dlpi_phdr",  ctypes.c_void_p),
+                    ("dlpi_phnum", ctypes.c_uint16)]
+
+
+# This function is called on each dynamically loaded library attached to our
+# process. It checks if any of this library matches the name given in data and
+# if a match is found, it set the global variable _MODULE_NAME to the full path
+# of this library, to be loaded with ctypes.CDLL.
+def check_module(info, size, data):
+    global _MODULE_NAME
+    # recast the name of the module as a string
+    desired_module = ctypes.cast(data, ctypes.c_char_p).value.decode('utf-8')
+
+    # Get the name of the current library
+    info = ctypes.cast(info, ctypes.POINTER(dl_phdr_info))
+    module_name = info.contents.dlpi_name
+
+    # If the current library is the one we are looking for, set the global
+    # variable with the desired path and return 1 to stop `dl_iterate_phdr`.
+    if module_name:
+        module_name = module_name.decode("utf-8")
+        if module_name.find(desired_module) >= 0:
+            _MODULE_NAME = module_name
+            return 1
```

Comment:
This is kind of unsafe it the module_name is not very specific. Maybe we should do something like:

```python
if module_path.rsplit('/', 1)[-1].startswith('lib' + module_name):
    _MODULE_PATH = module_path
    return 1
```